### PR TITLE
fix: Null cases for openai prompt and embeddings

### PR DESCRIPTION
**Pull request by ranadeepsingh**
## Related Issues/PRs

#2444 - caused the bug and divergence
#2456 - suggested partial fix

## What changes are proposed in this pull request?

Fix bug in OpenAIEmbedding when input is None/Null it was giving Null Pointer Exception.
Revamped and changed OpenAIEmbedding and OpenAIPrompt to return None/Null whne input is None/Null regardless of returnUsage is True or False.
+ Add tests for null case so this bug doesn't occur again

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change any dependencies?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure the dependencies are resolved correctly, and list changes here.

## Does this PR add a new feature? If so, have you added samples on website?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure you have added samples following below steps.

1. Find the corresponding markdown file for your new feature in `website/docs/documentation` folder.
   Make sure you choose the correct class `estimators/transformers` and namespace.
2. Follow the pattern in markdown file and add another section for your new API, including pyspark, scala (and .NET potentially) samples.
3. Make sure the `DocTable` points to correct API link.
4. Navigate to website folder, and run `yarn run start` to make sure the website renders correctly.
5. Don't forget to add `<!--pytest-codeblocks:cont-->` before each python code blocks to enable auto-tests for python samples.
6. Make sure the `WebsiteSamplesTests` job pass in the pipeline.

### CODE REVIEW

File: cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIEmbedding.scala [hunk 1]
```diff
@@ -101,13 +101,18 @@ class OpenAIEmbedding (override val uid: String) extends OpenAIServicesBase(uid)
...
         getOutputCol,
-        struct(
-          vectorCol.alias("response"),
-          usageCol.alias("usage")
+        when(responseCol.isNotNull,
+          struct(
+            vectorCol.alias("response"),
+            usageCol.alias("usage")
+          )
         )
```
[the remark this answers is not in the document]
Comment (ranadeepsingh) [reply cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIEmbedding.scala:109]: unnecessary

File: cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIEmbedding.scala [hunk 2]
```diff
@@ -101,13 +101,18 @@ class OpenAIEmbedding (override val uid: String) extends OpenAIServicesBase(uid)
...
       )
     } else {
-      parsed.withColumn(getOutputCol, vectorCol)
+      parsed.withColumn(
+        getOutputCol,
+        when(responseCol.isNotNull, vectorCol)
```
[the remark this answers is not in the document]
Comment (ranadeepsingh) [reply cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIEmbedding.scala:114]: unnecessary

File: cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIPrompt.scala [hunk 3]
```diff
@@ -277,10 +277,15 @@ class OpenAIPrompt(override val uid: String) extends Transformer
         val usageCol = normalizeUsageColumn(responseCol.getField("usage"), mapping)
         df.withColumn(
           getOutputCol,
-          F.struct(parsedCol.alias("response"), usageCol.alias("usage"))
+          F.when(parsedCol.isNotNull,
+            F.struct(parsedCol.alias("response"), usageCol.alias("usage"))
+          )
```
[the remark this answers is not in the document]
Comment (ranadeepsingh) [reply cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIPrompt.scala:282]: unnecessary

File: cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIPrompt.scala [hunk 4]
```diff
@@ -277,10 +277,15 @@ class OpenAIPrompt(override val uid: String) extends Transformer
...
         )
       case None =>
-        df.withColumn(getOutputCol, parsedCol)
+        df.withColumn(
+          getOutputCol,
+          F.when(parsedCol.isNotNull, parsedCol)
```
[the remark this answers is not in the document]
Comment (ranadeepsingh) [reply cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIPrompt.scala:287]: unnecessary

**Review by BrendanWalsh — APPROVED** [review]
